src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java

```diff
@@ -0,0 +1,58 @@
+package com.snowflake.kafka.connector.records;
+
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+
+import java.util.Map;
+
+public class SnowflakeMetadataConfig {
+  boolean createtimeFlag = true;
+  boolean topicFlag = true;
+  boolean offsetAndPartitionFlag = true;
+  boolean allFlag = true;
+
+  /**
+   * set flag to false only if metadata config is set to false in config
+   * @param config a String to String map of configs
+   */
+  public void parseMetadataConfig(Map<String, String> config)
+  {
+    // set all to be true to avoid bugs related with this function been called multiple times
+    createtimeFlag = true;
+    topicFlag = true;
+    offsetAndPartitionFlag = true;
+    allFlag = true;
+    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_CREATETIME) &&
+        !config.get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_CREATETIME)
+          .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT))
+    {
+      createtimeFlag = false;
+    }
+    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_TOPIC) &&
+      !config.get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_TOPIC)
+        .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT))
+    {
+      topicFlag = false;
+    }
+    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_OFFSET_AND_PARTITION) &&
+      !config.get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_OFFSET_AND_PARTITION)
+        .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT))
+    {
+      offsetAndPartitionFlag = false;
+    }
+    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_ALL) &&
+      !config.get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_ALL)
+        .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT))
+    {
+      allFlag = false;
+    }
+  }
+
+  public String toString()
+  {
+    return
```

Comment:
Why does the toString method return a JSON format record? Will we use it in logging or test only?

Reply:
I used that for testing. In the committed code this function is not called, so user won't see this.

Reply:
I see, and it is not a JSON format actually.